core/src/main/java/org/elasticsearch/common/network/NetworkModule.java

```diff
@@ -381,6 +395,37 @@ public void registerTaskStatus(String name, Writeable.Reader<? extends Task.Stat
         namedWriteableRegistry.register(Task.Status.class, name, reader);
     }
 
+    /**
+     * Register an allocation command.
+     * <p>
+     * This lives here instead of the more aptly named ClusterModule because the Transport client needs these to be registered.
```

Comment:
How about making AllocationCommandRegistry a proper module (that has a reference to NamedWriteableRegistry) and adding this method to that module?